orderbook/src/database/trades.rs

```diff
@@ -0,0 +1,305 @@
+                },
+                Event::DbTrade(DbTrade {
+                    order_uid: order_ids[1],
+                    sell_amount_including_fee: U256::from(3),
+                    buy_amount: U256::from(2),
+                    fee_amount: U256::from(2),
+                }),
+            ),
+            (
+                EventIndex {
+                    block_number: 2,
+                    log_index: 1,
+                },
+                Event::DbTrade(DbTrade {
+                    order_uid: order_ids[2],
+                    sell_amount_including_fee: U256::from(4),
+                    buy_amount: U256::from(3),
+                    fee_amount: U256::from(2),
+                }),
+            ),
+        ])
+        .await
+        .unwrap();
+
+        return (owners, order_ids, trades);
+    }
+
+    async fn assert_trades(db: &Database, filter: &TradeFilter, expected: &[Trade]) {
+        let filtered = db
+            .trades(&filter)
+            .try_collect::<HashSet<Trade>>()
+            .await
+            .unwrap();
+        let expected = expected.iter().cloned().collect::<HashSet<_>>();
+        assert_eq!(filtered, expected);
+    }
+
+    #[tokio::test]
+    #[ignore]
+    async fn postgres_trades_with_and_without_filter() {
+        let db = Database::new("postgresql://").unwrap();
+        db.clear().await.unwrap();
+
+        let (owners, order_ids, trades) = populate_dummy_trade_db(db.clone()).await;
+
+        // No filter assertion
+        assert_trades(&db, &TradeFilter::default(), &trades).await;
+
+        // Owner Filter assertions
+        assert_trades(
+            &db,
+            &TradeFilter {
+                owner: Some(owners[1]),
+                ..Default::default()
+            },
+            &[],
+        )
+        .await;
+
+        assert_trades(
+            &db,
+            &TradeFilter {
+                owner: Some(owners[0]),
+                ..Default::default()
+            },
+            &trades[0..1],
+        )
+        .await;
+
+        assert_trades(
+            &db,
+            &TradeFilter {
+                owner: Some(owners[2]),
+                ..Default::default()
+            },
+            &trades[1..],
+        )
+        .await;
+
+        // Order ID filter assertions
+        assert_trades(
+            &db,
+            &TradeFilter {
+                order_uid: Some(order_ids[0]),
+                ..Default::default()
+            },
+            &[],
+        )
+        .await;
+
+        assert_trades(
+            &db,
+            &TradeFilter {
+                order_uid: Some(order_ids[1]),
+                ..Default::default()
+            },
+            &trades[0..1],
+        )
+        .await;
+
+        assert_trades(
+            &db,
+            &TradeFilter {
+                order_uid: Some(order_ids[2]),
+                ..Default::default()
+            },
+            &trades[1..],
+        )
+        .await;
+    }
+}
```

Comment:
I would like to see a short test for the case where for whatever reason we have a trade event but not matching order uid to make sure that the query still works and does not include the trade.

Reply:
I believe we have a test that explicitly returns nothing in the response a when filtering by valid order_uid if this is what you are suggesting.

Reply:
I mean that we have a trade event for uid X, but no corresponding order with the same uid. So what should happen is that during the join of trades and orders this trade gets skipped because there is no corresponding order.

Reply:
Interesting, we can test this but I would hope that this never happens!

Reply:
Included!